Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.35109&attempt_id=4444e774-4bd0-412d-8a09-ae644346cc75)

<details><summary>Partial log (click to expand)</summary><p>

```
patching sources
configuring
no configure script, doing nothing
building
build flags: SHELL=/nix/store/19cc89zq5n68ibr53izy4zgcr5qg2rza-bash-4.4-p12/bin/bash
gcc -lnet -o arpoison `libnet-config --defines` -I/opt/local/include -L/opt/local/lib arpoison.c
/nix/store/19cc89zq5n68ibr53izy4zgcr5qg2rza-bash-4.4-p12/bin/bash: gcc: command not found
make: *** [Makefile:5: arpoison] Error 127
builder for '/nix/store/fg4fx41nw94ni4mps4w08vwam23sz8ix-arpoison-0.7.drv' failed with exit code 2
error: build of '/nix/store/fg4fx41nw94ni4mps4w08vwam23sz8ix-arpoison-0.7.drv' failed
```
</p></details>

--- a/pkgs/tools/networking/arpoison/default.nix
+++ b/pkgs/tools/networking/arpoison/default.nix
@@ -10,6 +10,8 @@ stdenv.mkDerivation rec {
     sha256 = "0krhszx3s0qwfg4rma5a51ak71nnd9xfs2ibggc3hwiz506s2x37";
   };
 
+  makeFlags = ["CC=cc"];
+
   installPhase = ''
     mkdir -p $out/bin $out/share/man/man8
     gzip arpoison.8

--- a/pkgs/tools/networking/arpoison/default.nix
+++ b/pkgs/tools/networking/arpoison/default.nix
@@ -10,7 +10,7 @@ stdenv.mkDerivation rec {
     sha256 = "0krhszx3s0qwfg4rma5a51ak71nnd9xfs2ibggc3hwiz506s2x37";
   };
 
-  makeFlags = ["CC=cc"];
+  postPatch = "substituteInPlace Makefile --replace gcc cc";
 
   installPhase = ''
     mkdir -p $out/bin $out/share/man/man8

